docs/source/reference/kubernetes/kubernetes-setup.rst

```diff
@@ -0,0 +1,223 @@
+.. _kubernetes-setup:
+
+Kubernetes Cluster Setup
+========================
+
+
+.. note::
+    This is a guide for cluster administrators on how to setup Kubernetes clusters
+    for use with SkyPilot.
+
+    If you are a SkyPilot user and your cluster administrator has already setup a cluster
+    and shared a kubeconfig file with you, :ref:`Submitting tasks to Kubernetes <kubernetes-instructions>`
+    explains how to submit tasks to your cluster.
+
+
+SkyPilot on Kubernetes is designed to work with most Kubernetes distributions and deployment environments.
+
+To connect to a Kubernetes cluster, SkyPilot needs:
+
+* An existing Kubernetes cluster running Kubernetes v1.20 or later.
+* A `Kubeconfig <kubeconfig>`_ file containing access credentials and namespace to be used.
+
+
+Deployment Guides
+-----------------
+Below we show minimal examples to setup a new Kubernetes cluster in different environments, including hosted services on the cloud, and generating kubeconfig files which can be :ref:`used by SkyPilot <kubernetes-instructions>`.
+
+..
+  TODO(romilb) - Add a table of contents/grid cards for each deployment environment.
+
+.. _kubernetes-setup-kind:
+
+Deploying locally on your Laptop
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to try out SkyPilot on Kubernetes on your laptop or run SkyPilot
+tasks locally without requiring any cloud access, we provide the
+:code:`sky local up` CLI to create a 1-node Kubernetes cluster locally.
+
+Under the hood, :code:`sky local up` uses `kind <https://kind.sigs.k8s.io/>`_,
+a tool for creating a Kubernetes cluster on your local machine.
+It runs a Kubernetes cluster inside a container, so no setup is required.
+
+1. Install `Docker <https://docs.docker.com/engine/install/>`_ and `kind <https://kind.sigs.k8s.io/>`_.
+2. Run :code:`sky local up` to launch a Kubernetes cluster and automatically configure your kubeconfig file:
+
+    .. code-block:: console
```

Comment:
Hmm, on my local these codeblocks rendered in blockquotes:

<img width="821" alt="Screen Shot 2023-09-11 at 22 22 46" src="https://github.com/skypilot-org/skypilot/assets/592670/91be6c25-16eb-4732-a5a7-9620475e524d">

Reply:
Ah good catch - I've fixed this now:
<img width="859" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/b681853e-b7a7-4d88-982e-4d5bf9fa1671">